napari/layers/graph/graph.py

```diff
@@ -399,10 +400,26 @@ def add(
         coords : sequence of indices to add point at
         indices : optional indices of the newly inserted nodes.
         """
+        self.events.data(
+            value=self.data,
+            action=ActionType.ADDING,
+            data_indices=(-1,),
+            vertex_indices=((),),
+        )
+
         prev_size = self.data.n_allocated_nodes
         self.data.add_nodes(indices=indices, coords=coords)
```

Comment:
Indices could be `None` if you're just adding to the end. `self.data.add_nodes` returns the indices added. We may not know what indices we're changing at `ADDING` time, if there's any empty spaces in the buffer. We may need API for this to get from `Graph` object the indices it will add